openapi/generator/templates/okta.go.hbs

```diff
@@ -28,9 +30,15 @@ func NewClient(config *Config) *Client {
 		config = NewConfig()
 	}
 
+	var oktaCache cache.Cache
+	oktaCache = cache.NewGoCache(config.Okta.Client.Cache.DefaultTtl, config.Okta.Client.Cache.DefaultTti)
+	if config.Okta.Client.Cache.Enabled == false {
+		oktaCache = cache.NewNoOpCache()
+	}
+
```

Comment:
I'm not sure how I would structure this in Go, but it would be good to reduce the amount of code that we have in this template.  For the Node SDK we achieved this by doing all the client building configuration outside of the generated client ([client.js](https://github.com/okta/okta-sdk-nodejs/blob/master/src/client.js)) and then it extends the generated client ([https://github.com/okta/okta-sdk-nodejs/blob/master/src/generated-client.js]()).

Reply:
I generate the okta.go file for reasons of generating some of the properties for the client from the spec